### PR TITLE
Add support for the qemuriscv32 machine

### PR DESCRIPTION
**Pull request by alistair23**
Add support for the 32-bit QEMU machine.

### CODE REVIEW

File: recipes-core/glibc/glibc-initial_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Force it to use the RISC-V fork
+
+GLIBC_GIT_URI_qemuriscv32 = "git://github.com/riscv/riscv-glibc.git"
+SRCBRANCH_qemuriscv32 = "32bit"
+SRCREV_qemuriscv32 = "83d5eaa75361ba0cb7969669452c9ff233ac710b"
+
+SRC_URI_remove_qemuriscv32 = " \
+    file://0031-sysdeps-ieee754-prevent-maybe-uninitialized-errors-w.patch \
+    file://0032-sysdeps-ieee754-soft-fp-ignore-maybe-uninitialized-w.patch \
+    file://0033-locale-prevent-maybe-uninitialized-errors-with-Os-BZ.patch \
```
Comment (kraj) [inline recipes-core/glibc/glibc-initial_%.bbappend]: I dont see these patches added, where will they come from ?

File: conf/machine/qemuriscv32.conf [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#@TYPE: Machine
```
Comment (kraj) [inline conf/machine/qemuriscv32.conf]: can we create a ```riscv-common.inc``` and move common bits into it from 32bit and 64bit conf files.

File: recipes-core/glibc/glibc-initial_%.bbappend (hunk 1 above)
Comment (alistair23) [reply recipes-core/glibc/glibc-initial_%.bbappend]: These are being removed, or at least not applied as they don't apply to the newer tree needed for 32bit support

File: conf/machine/qemuriscv32.conf (hunk 2 above)
Comment (alistair23) [reply conf/machine/qemuriscv32.conf]: Yep

Comment (alistair23) [reply conf/machine/qemuriscv32.conf]: Updated in the new version

**Pull request merged**